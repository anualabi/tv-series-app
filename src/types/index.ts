export interface SeriesProps {
    Title: string;
    Year?: string;
    Rated?: string;
    Released: string;
    Runtime?: string;
    Genre?: string;
    Director?: string;
    Writer?: string;
    Actors?: string;
    Plot: string;
    Language?: string;
    Country?: string;
    Awards?: string;
    Poster: string;
    Ratings?: [
        {
            Source: string;
            Value: string;
        }
    ],
    Metascore?: string;
    imdbRating: string;
    imdbVotes?:string;
    imdbID?: string;
    Type?: string;
    totalSeasons: string;
    Response?: string;
}

export type EpisodesProps = {
    Title: string;
    Plot: string;
    Released: string;
    Episode: string;
    Poster: string;
    imdbRating: string;
    imdbID: string;
}[]

export interface SeasonProps {
    Title: string;
    Plot?: string;
    Season: string;
    totalSeasons: string;
    Episodes: EpisodesProps,
    Response: string;
}

export interface EpisodeProps {
    Title: string;
    Year?: string;
    Rated?: string;
    Released: string;
    Season?: string;
    Episode: string;
    Runtime?: string;
    Genre?: string;
    Director?: string;
    Writer?: string;
    Actors?: string;
    Plot: string;
    Language?: string;
    Country?: string;
    Awards?: string;
    Poster: string;
    Ratings?: [
        {
            Source: string;
            Value: string;
        }
    ],
    Metascore?: string;
    imdbRating: string;
    imdbVotes?:string;
    imdbID?: string;
    seriesID?: string;
    Type?: string;
    Response?: string;
}
