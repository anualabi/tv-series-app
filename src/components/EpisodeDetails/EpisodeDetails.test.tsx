import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import EpisodeDetail from './EpisodeDetails';
import { EpisodeProps } from '@/types'

const mockProps = {
    Title: 'Better-Like',
    Plot: 'Test plot description for the episode.',
    Poster: '/assets/episode-one.svg',
    Episode: '1',
    Released: '2023-08-10',
    imdbRating: '8.5',
};

const customRender = (props: EpisodeProps) => render(<EpisodeDetail data={props} />);

describe('EpisodeDetail', () => {
    test('should render snapshot', () => {
        const { asFragment } = customRender({...mockProps})
        
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the image with the correct alt text', () => {
        customRender({...mockProps})
        
        const imgElement = screen.getByAltText('Better-Like');
        
        expect(imgElement).toBeInTheDocument();
    });

    test('renders the episode details correctly', () => {
        customRender({ ...mockProps })
        
        expect(screen.getByText('Episode 1 â€” 2023-08-10')).toBeInTheDocument();
        expect(screen.getByText('Better-Like')).toBeInTheDocument();
        expect(screen.getByText('Test plot description for the episode.')).toBeInTheDocument();
        expect(screen.getByText('8.5')).toBeInTheDocument();
    });
});
