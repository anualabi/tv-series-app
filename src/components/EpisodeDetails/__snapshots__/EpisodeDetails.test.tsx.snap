// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeDetail should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full h-full"
  >
    <div
      class="h-3/5 md:h-2/3 flex-grow relative"
    >
      <img
        alt="Better-Like"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        src="/assets/episode-one.svg"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </div>
    <div
      class="h-2/5 md:h-1/3 bg-white text-black"
    >
      <div
        class="flex justify-between items-center h-1/3 p-4 md:p-8 border-b border-gray-200"
      >
        <p
          class="text-base xl:text-lg pt-1.5"
        >
          Episode 1 â€” 2023-08-10
        </p>
        <div
          class="text-base xl:text-lg flex items-center"
        >
          <svg
            height="26.62"
            viewBox="0 0 27.985 26.615"
            width="27.98"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.992,0l4.324,8.761,9.669,1.405-7,6.819,1.651,9.63-8.647-4.547L5.345,26.615,7,16.986,0,10.166,9.669,8.761Z"
              fill="#efd358"
              id="Shape"
            />
          </svg>
          <p
            class="ml-3.5 pt-1.5"
          >
            <span
              class="font-bold"
            >
              8.5
            </span>
            /10
          </p>
        </div>
      </div>
      <div
        class="h-2/3 flex flex-col justify-center p-4 md:p-8"
      >
        <h2
          class="text-2xl xl:text-3xl font-bold mb-2"
        >
          Better-Like
        </h2>
        <p
          class="text-base md:text-lg xl:text-xl line-clamp-5 lg:line-clamp-2"
        >
          Test plot description for the episode.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
